llvm/include/llvm/ADT/STLExtras.h

```diff
@@ -2002,6 +2002,13 @@ template <typename R, typename Compare> auto max_element(R &&Range, Compare C) {
   return std::max_element(adl_begin(Range), adl_end(Range), C);
 }
 
+/// Provide wrappers to std::mismatch which take ranges instead of having to
+/// pass begin/end explicitly.
+template <typename R1, typename R2> auto mismatch(R1 &&Range1, R2 &&Range2) {
```

Comment:
Could you add a comment of explanation for what this algorithm does? (and maybe an example too) I think that this is a lesser known std function

Reply:
Also, do we want to support the overload that takes a binary predicate?

Reply:
I don't intend to use that overload. My understanding is that they are added on-demand. But I can add it.

Reply:
I think it's fine either way

Reply:
SG, then I'll just add what I intend to use. I added a comment with an example. PTAL.